sdks/go/pkg/beam/core/typex/special.go

```diff
@@ -64,6 +65,23 @@ type Window interface {
 	Equals(o Window) bool
 }
 
+type Timing string
```

Comment:
Instead of using a string, consider using an `int` (or even a uint8) as the base type. It's more compact in memory. It also lets one avoid doing string comparisons to know values for encodings, and avoids some awkward conversions that this PR is currently making.

See https://yourbasic.org/golang/iota/ for a good rundown for how to do "enums" in Go. As well as in Effective Go. https://golang.org/doc/effective_go#constants

Reply:
Thanks for this! This is really helpful.